streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java

```diff
@@ -254,4 +256,25 @@ public TaskType taskType() {
     public String changelogFor(final String storeName) {
         return stateManager().changelogFor(storeName);
     }
+
+    @Override
+    public void addProcessorMetadataKeyValue(final String key, final long value) {
+        processorMetadata.addMetadata(key, value);
+    }
+
+    @Override
+    public Long getProcessorMetadataForKey(final String key) {
+        return processorMetadata.getMetadata(key);
+    }
+
+    @Override
+    public void setProcessorMetadata(final ProcessorMetadata metadata) {
```

Comment:
Why do we need both "per key" and "full" getters/setters -- could we limit it to only one of both?

Reply:
This is used in restoration to set processor metadata as a whole after deserialization and merging. The per key setter is used by processor. Maybe we can get rid of `getProcessorMetadata` if it's not used.